pkg/controller/statefulset/config/v1alpha1/defaults.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	kubectrlmgrconfigv1alpha1 "k8s.io/kube-controller-manager/config/v1alpha1"
+)
+
+// RecommendedDefaultStatefulSetControllerConfiguration defaults a pointer to a
+// StatefulSetControllerConfiguration struct. This will set the recommended default
+// values, but they may be subject to change between API versions. This function
+// is intentionally not registered in the scheme as a "normal" `SetDefaults_Foo`
+// function to allow consumers of this type to set whatever defaults for their
+// embedded configs. Forcing consumers to use these defaults would be problematic
+// as defaulting in the scheme is done as part of the conversion, and there would
+// be no easy way to opt-out. Instead, if you want to use this defaulting method
+// run it in your wrapper struct of this type in its `SetDefaults_` method.
+func RecommendedDefaultStatefulSetControllerConfiguration(obj *kubectrlmgrconfigv1alpha1.StatefulSetControllerConfiguration) {
+	if obj.ConcurrentStatefulSetSyncs == 0 {
+		obj.ConcurrentStatefulSetSyncs = 1
```

Comment:
if the current default is causing issues, should we bump this to 5? @smarterclayton @kubernetes/sig-apps-misc was there a reason statefulsets were set to 1 instead of 5?

Reply:
per https://github.com/kubernetes/kubernetes/issues/76027 the RecommendedDefault* pattern isn't required for these single-use defaults, just define a normal defaulting method for the `StatefulSetControllerConfiguration` type

Reply:
> if the current default is causing issues, should we bump this to 5? @smarterclayton @kubernetes/sig-apps-misc was there a reason statefulsets were set to 1 instead of 5?

Done as your suggestion. We have changed the num of concurrent workers to 16 and disable resync for years in our clusters, and works well. Also want to know if there is any risk to change the default value  @smarterclayton